tests/conftest.py

```diff
@@ -542,3 +542,100 @@ def test_user(test_client, runestone_db_tools):
 def test_user_1(runestone_db_tools, test_user):
     runestone_db_tools.create_course('test_course_1')
     return test_user('test_user_1', 'password_1', 'test_course_1')
+
+
+class _TestAssignment(object):
+    assignment_count = 0
+    def __init__(self, test_client, test_user, runestone_db_tools, aname, course_name):
+        self.test_client = test_client
+        self.runestone_db_tools = runestone_db_tools
+        self.assignment_name = aname
+        self.course_name = course_name
+        self.description = "default description"
+        self.is_visible = False
+        self.due = datetime.datetime.utcnow() + datetime.timedelta(days=7)
+        self.assignment_instructor = test_user('assign_instructor_{}'.format(_TestAssignment.assignment_count),
+            'password', course_name)
+        self.assignment_instructor.make_instructor()
+        self.assignment_instructor.login()
+        self.assignment_id = json.loads(
+            self.test_client.validate('admin/createAssignment',
+                                    data={'name': self.assignment_name})
+        )[self.assignment_name]
+        assert self.assignment_id
+        _TestAssignment.assignment_count += 1
+
+
+    def addq_to_assignment(self, **kwargs):
+        if 'points' not in kwargs:
+            kwargs['points'] = 1
+        kwargs['assignment'] = self.assignment_id
+        assert self.test_client.validate(
+            'admin/add__or_update_assignment_question', data=kwargs
+        ) != json.dumps('Error')
```

Comment:
Seems fragile -- is there any way to measure this as an equality (verify that success or something similar was returned)? Here, even a return of `error` would still give a test passed.

Reply:
I copied that straight out of your code.  :smile:

Reply:
The success indicator is that a dictionary of values is returned.  I could modify the dictionary to have a status key.  I was trying to keep this code simple and let the actual test take care of details.  For example the keys returned are total, activity_count, and activities_required.  That doesn't seem like something this method ought to worry about.

Reply:
OK, I added a simple status key in both cases to indicate success.  No more `!=`

Reply:
Cool!